scala-package/examples/src/test/scala/org/apache/mxnetexamples/gan/GanExampleSuite.scala

```diff
@@ -44,7 +44,9 @@ class GanExampleSuite extends FunSuite with BeforeAndAfterAll{
 
         val context = Context.gpu()
 
-        val output = GanMnist.runTraining(modelDirPath, context, modelDirPath, 5)
+        val output = NDArrayCollector.auto().withScope {
```

Comment:
I don't think you need NDCollectors here. It should be sufficient to just have it inside the training loop(for each epoch)